src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +34,54 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+
+        //splits address into 4 different components
+        String[] splitAddress = splitAddress(trimmedAddress);
+
+        block.setData(splitAddress[0]);
+        street.setData(splitAddress[1]);
+        unit.setData(splitAddress[2]);
+        postalCode.setData(splitAddress[3]);
+
+        String temp = block.getData();
+
+        if (street.getData() != null) {
+            temp = temp.concat(" " + street.getData());
+        }
+        if (unit.getData() != null) {
+            temp = temp.concat(" " + unit.getData());
+        }
+        if (postalCode.getData() != null) {
+            temp = temp.concat(" " + postalCode.getData());
+        }
+
+        value = temp;
+    }
+
+    /**
+     * splits a given address into four Strings representing block, street, unit, postal code
+     * first 3 fields contains the comma delimiter
+     * @param  address input by user
+     * @return split address
+     */
```

Comment:
There should be a blank line separates description and `@param` & `@return` section. 

`splits` also need to be capitalized as `Splits` because comments should follow proper grammar.